cmd/thanos/compact.go

```diff
@@ -92,6 +92,9 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application, name stri
 	maxCompactionLevel := cmd.Flag("debug.max-compaction-level", fmt.Sprintf("Maximum compaction level, default is %d: %s", compactions.maxLevel(), compactions.String())).
 		Hidden().Default(strconv.Itoa(compactions.maxLevel())).Int()
 
+	metaSyncConcurrency := cmd.Flag("meta-sync-concurrency", "Number of goroutines to use when syncing metadata from object storage.").
+		Default("1").Int()
```

Comment:
We can actually make it faster by default to some reasonable number like 20. (same as store gateway). There is no reason, NOT to.